src/digitalPins.cpp

```diff
@@ -0,0 +1,41 @@
+#include "digitalPins.h"
+#include "const.h"
+#include <Arduino.h>
+
+#include "STM32TimerInterrupt_Generic.h"
+
+// Ticker to poll input readings at fixed rate
+STM32TimerInterrupt digitalReader(TIM7);
+
```

Comment:
You need to instantiate a `volatile Digital_Data digital_data` here since it was declared extern in the `.h` file.